include/IECoreScene/Primitive.inl

```diff
@@ -38,6 +38,75 @@
 namespace IECoreScene
 {
 
+template<typename T>
+std::unique_ptr< PrimitiveVariable::IndexedView<T> >
+Primitive::variableIndexedView( const std::string &name, PrimitiveVariable::Interpolation requiredInterpolation, bool throwIfInvalid )
+{
+	PrimitiveVariableMap::const_iterator it = variables.find( name );
+	if( it == variables.end() )
+	{
+		if ( throwIfInvalid )
+		{
+			throw IECore::InvalidArgumentException( boost::str( boost::format( "Primitive::variableIndexedView - No primvar named '%1%' found" ) % name ) );
+		}
+		else
+		{
+			return nullptr;
+		}
+	}
+
+	if (!it->second.interpolation == requiredInterpolation)
+	{
+		if ( throwIfInvalid )
+		{
+			throw IECore::InvalidArgumentException(
+				boost::str(
+					boost::format(
+						"Primitive::variableIndexedView - PrimVar '%1%' interpolation (%2%) doesn't match requiredInterpolation (%3%)"
+					) % name % it->second.interpolation % requiredInterpolation
+				)
+			);
+		}
+		else
+		{
+			return nullptr;
+		}
+	}
+
+	auto makeIndexedView = [&it]() -> std::unique_ptr< PrimitiveVariable::IndexedView<T> >
+	{
+		return std::unique_ptr< PrimitiveVariable::IndexedView<T> > ( new PrimitiveVariable::IndexedView<T>( it->second ) );
+	};
+
+	if ( throwIfInvalid )
+	{
+		return makeIndexedView();
+	}
+	else
+	{
+		try
```

Comment:
Rather than using exceptions for control flow, how about we just use the second IndexedView constructor instead?

```
// I'm taking some liberties with the return type - I'll discuss that in detail elsewhere.

typedef TypedData<vector<T>> DataType;
const DataType *data = runTimeCast<const DataType>( it->second.data.get() );
if( data )
{
     return IndexedView( data->readable(), it->second.indices ? it->second.indices->readable() : nullptr );
}
else if( throwIfInvalid )
{
    thow ...
}
else
{
    return boost::none;
}
```

Reply:
Perhaps it would be easier to use the TypedData as the template parameter as I won't have to jump through hoops to work out the TypedData from the value type.

Reply:
Yeah, although I think looking at the call sites as you suggested below is more important - better to make it convenient for client code...